### PR TITLE
1617: Optional pipeline name

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

In the past the store imports were managed by backend pipelines for every project. These imports ran with scheduled cron jobs.
For the sozialpass projects we will use the store import module in the administration gui, so there will be no need for a backend import pipeline, so this value should be optional
### Proposed changes

- make pipeline name optional
- add a null check before starting the import pipelines for each project
- remove pipeline name for koblenz from config.yml (this should also happen in salt)

### Side effects

- no

### Testing

- run /backend/import  and check if the other pipelines still run successful
- for koblenz should be displayed that there is no store import needed (no pipeline should be defined in your local config.yml)

### Resolved issues

Fixes: #1617

### CODE REVIEW

File: backend/src/main/resources/config/config.yml [hunk 1]
```diff
@@ -58,7 +58,6 @@ projects:
       password: OVERRIDE_IN_LOCAL_CONFIG
   - id: showcase.entitlementcard.app
     importUrl: https://example.com
-    pipelineName: BerechtigungskarteShowcase
```
Comment (steffenkleinle) [inline backend/src/main/resources/config/config.yml:61]: I think you forgot to remove `pipelineName: SozialpassKoblenz` in L49.

**Review by steffenkleinle — COMMENTED** [review]
Seems to work correctly once `pipelineName: SozialpassKoblenz` in config.yml L49 is removed.

File: backend/src/main/resources/config/config.yml (hunk 1 above)
Comment (f1sh1918) [reply backend/src/main/resources/config/config.yml:61]: you are right. thx

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/Importer.kt [hunk 2]
```diff
@@ -10,6 +10,11 @@ object Importer {
...
         val project = config.findProject()
 
+        if (project.pipelineName == null) {
+            logger.info("There is no store import pipeline for ${project.id}")
+            return true
+        }
+
         return try {
             logger.info("== Pipeline ${project.pipelineName} started ==")
             when (project.pipelineName) {
```
Comment (ztefanie) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/Importer.kt:20]: Shouldn't Nuernberg be deleted in line 17? in the description of the pr you write "For the sozialpass projects we will use the store import module in the administration gui". So I expect this changes to apply to Nuernberg and Koblenz.

**Comment by ztefanie** [comment]
> i get this error ( i am on the latest version of this current branch and on a fresh database):
> 
> ```
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - The following changes were made to the database:
> Count stores deleted:   0
> Count stores created:   121
> Count stores untouched: 0
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - == Pipeline SozialpassNuernberg successfully finished ==
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - == Pipeline SozialpassKoblenz started ==
> Exception in thread "main" java.lang.Error: Invalid pipeline name 'SozialpassKoblenz'!
> 	at app.ehrenamtskarte.backend.stores.importer.Importer.import(Importer.kt:24)
> 	at app.ehrenamtskarte.backend.Import.run(EntryPoint.kt:92)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
> 	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
> 	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
> 	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
> 	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
> 	at app.ehrenamtskarte.backend.EntryPointKt.main(EntryPoint.kt:162)
> ```

**Comment by f1sh1918** [comment]
> i get this error:
> 
> ```
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - The following changes were made to the database:
> Count stores deleted:   0
> Count stores created:   121
> Count stores untouched: 0
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - == Pipeline SozialpassNuernberg successfully finished ==
> [main] INFO app.ehrenamtskarte.backend.stores.importer.Importer - == Pipeline SozialpassKoblenz started ==
> Exception in thread "main" java.lang.Error: Invalid pipeline name 'SozialpassKoblenz'!
> 	at app.ehrenamtskarte.backend.stores.importer.Importer.import(Importer.kt:24)
> 	at app.ehrenamtskarte.backend.Import.run(EntryPoint.kt:92)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
> 	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
> 	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
> 	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
> 	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
> 	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
> 	at app.ehrenamtskarte.backend.EntryPointKt.main(EntryPoint.kt:162)
> ```

@ztefanie this happens due your pipelineName in `config.local.yml` you have to remove this entry for the koblenz project

**Review by ztefanie — APPROVED** [review]
Makes sense :D

**Pull request merged**